diff --git a/pkg/_macros/lib/src/executor/augmentation_library.dart b/pkg/_macros/lib/src/executor/augmentation_library.dart
index 71489b9aa98..f776aa3c438 100644
--- a/pkg/_macros/lib/src/executor/augmentation_library.dart
+++ b/pkg/_macros/lib/src/executor/augmentation_library.dart
@@ -290,7 +290,7 @@ class _Builder {
           (OmittedTypeAnnotation omittedType, _SynthesizedNamePart part) {
         String name = '$prefix${index++}';
         part.text = name;
-        _omittedTypes[omittedType] = name;
+        _omittedTypes![omittedType] = name;
       });
     }
 
