diff --git a/pkg/_macros/lib/src/executor/augmentation_library.dart b/pkg/_macros/lib/src/executor/augmentation_library.dart
index 71489b9aa98..f776aa3c438 100644
--- a/pkg/_macros/lib/src/executor/augmentation_library.dart
+++ b/pkg/_macros/lib/src/executor/augmentation_library.dart
@@ -290,7 +290,7 @@ class _Builder {
           (OmittedTypeAnnotation omittedType, _SynthesizedNamePart part) {
         String name = '$prefix${index++}';
         part.text = name;
-        _omittedTypes[omittedType] = name;
+        _omittedTypes![omittedType] = name;
       });
     }

diff --git a/pkg/vm/lib/modular/target/flutter.dart b/pkg/vm/lib/modular/target/flutter.dart
index 3a797cac..4010ce9d 100644
--- a/pkg/vm/lib/modular/target/flutter.dart
+++ b/pkg/vm/lib/modular/target/flutter.dart
@@ -9,11 +9,18 @@ import 'package:kernel/target/targets.dart';
 import 'package:kernel/transformations/track_widget_constructor_locations.dart';
 import 'package:vm/modular/target/vm.dart' show VmTarget;
 
+abstract class FlutterProgramTransformer {
+  void transform(Component component);
+}
+
 class FlutterTarget extends VmTarget {
   FlutterTarget(TargetFlags flags) : super(flags);
 
   late final WidgetCreatorTracker _widgetTracker = WidgetCreatorTracker();
 
+  static List<FlutterProgramTransformer> _flutterProgramTransformers = [];
+  static List<FlutterProgramTransformer> get flutterProgramTransformers => _flutterProgramTransformers;
+
   @override
   String get name => 'flutter';
 
@@ -58,6 +65,14 @@ class FlutterTarget extends VmTarget {
       DiagnosticReporter diagnosticReporter,
       {void Function(String msg)? logger,
       ChangedStructureNotifier? changedStructureNotifier}) {
+
+    if (_flutterProgramTransformers.length > 0) {
+      int flutterProgramTransformersLen = _flutterProgramTransformers.length;
+      for (int i=0; i<flutterProgramTransformersLen; i++) {
+        _flutterProgramTransformers[i].transform(component);
+      }
+    }
+
     super.performPreConstantEvaluationTransformations(
         component, coreTypes, libraries, diagnosticReporter,
         logger: logger, changedStructureNotifier: changedStructureNotifier);
